db/output_validator.h

```diff
@@ -0,0 +1,50 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+#pragma once
+#include "rocksdb/status.h"
+#include "test_util/sync_point.h"
+#include "util/hash.h"
+
+namespace rocksdb {
+// A class that validates key/value that is inserted to an SST file.
+// Pass every key/value of the file using OutputValidator::Add()
+// and the class validates key order and optionally calculate a hash
+// of all the key and value.
+class OutputValidator {
+ public:
+  explicit OutputValidator(const InternalKeyComparator& _icmp,
+                           bool _enable_order_check, bool _enable_hash) :
+  icmp_(_icmp), enable_order_check_(_enable_order_check), enable_hash_(_enable_hash) {}
+  Status Add(const Slice& key, const Slice& value) {
+    if (enable_hash_) {
```

Comment:
Can we move the implementation into a C++ file instead of a header and clean up the includes?